pretyping/detyping.ml

```diff
@@ -694,6 +694,13 @@ and detype_binder (lax,isgoal as flags) bk avoid env sigma na body ty c =
       let t = if s != InProp  && not !Flags.raw_print then None else Some (detype (lax,false) avoid env sigma ty) in
       GLetIn (na', c, t, r)
 
+and detype_array flags avoid env sigma args =
+  let ans = ref [] in
+  for i = Array.length args - 1 downto 0 do
+    ans := detype flags avoid env sigma args.(i) :: !ans;
+  done;
+  !ans
```

Comment:
Am I missing something or did you just unfold `CArray.map_to_list` into imperative style? If the latter, it would be better to actually rewrite `CArray.map_to_list` that way, so that the handful of other user of this function benefit from the optimization too. In fact, I might even let you get away with using `unsafe_get` there if you wish to ;-).